packages/edit-site/src/components/global-styles/font-library-modal/index.js

```diff
@@ -44,11 +46,23 @@ function FontLibraryModal( {
 	defaultTabId = 'installed-fonts',
 } ) {
 	const { collections, setNotice } = useContext( FontLibraryContext );
+	const canUserCreate = useSelect( ( select ) => {
+		const { canUser } = select( coreStore );
+		return canUser( 'create', 'font-families' );
+	}, [] );
+	const fontUploadsEnabled = useSelect( ( select ) => {
+		const { getEditorSettings } = select( editorStore );
+		return getEditorSettings().fontUploadsEnabled;
+	}, [] );
+
+	const tabs = [ DEFAULT_TAB ];
 
-	const tabs = [
-		...DEFAULT_TABS,
-		...tabsFromCollections( collections || [] ),
-	];
+	if ( canUserCreate && fontUploadsEnabled ) {
+		tabs.push( UPLOAD_TAB );
+		// In the future, font faces can be configured to use a remote url rather than a file upload as the font src property.
+		// In that case, collections tabs should still be shown when font uploads are disabled.
```

Comment:
Inline commenting standards for multiline comments https://developer.wordpress.org/coding-standards/inline-documentation-standards/javascript/#inline-comments

```suggestion
		/*
		 * In the future, font faces can be configured to use a remote url rather than a file upload as the font src property.
		 * In that case, collections tabs should still be shown when font uploads are disabled.
		 */
```